types/index.d.ts

```diff
@@ -0,0 +1,160 @@
+import type { AedesPacket } from 'aedes-packet';
+import type { IncomingMessage } from 'http';
+import type { Socket } from 'net';
+import type { Duplex, Readable } from 'stream';
+import type { EventEmitter } from 'events';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId?: string;
+  topic: string;
+  qos: number;
+}
+
+export interface Brokers {
+  [brokerId: string]: {
+    brokerId: string;
+  };
+}
+
+type Connection = Duplex | Socket;
+
+interface Client extends EventEmitter {
+  id: string;
+  clean: boolean;
+  version: number;
+  conn: Connection;
+  req?: IncomingMessage;
+  connecting: boolean;
+  connected: boolean;
+  closed: boolean;
+}
+
+declare class MemoryPersistence {
+  private _retained: any[];
```

Comment:
isnt this an AedesPacket array?

Reply:
Yes it seems but i was not sure where it was called in aedes. So that was type laziness ;)